utils/src/main/scala/za/co/absa/enceladus/utils/transformations/DeepArrayTransformations.scala

```diff
@@ -62,14 +190,105 @@ object DeepArrayTransformations {
       name
     }
 
-    // Returns true if a path consists only of 1 element meaning it is the leaf element of the input column path requested by the caller
-    def isLeafElement(path: Seq[String]): Boolean = path.lengthCompare(2) < 0 // More efficient version of path.length == 1
+    def ensureOutputColumnNameValid(columnName: String): Unit= {
```

Comment:
This method feels weird. It uses 3 variables, where two are input parameters of the encompassing function. The other could provide some flexibility but actually is always  `outputComunName`  a value fixated from another of the input parameters of the higher context as the first step there.